packages/flutter/lib/src/material/circle_avatar.dart

```diff
@@ -67,13 +69,19 @@ class CircleAvatar extends StatelessWidget {
   /// The color with which to fill the circle. Changing the background
   /// color will cause the avatar to animate to the new color.
   ///
-  /// If a background color is not specified, the theme's primary color is used.
+  /// If a background color is not specified, the theme's light primary color
+  /// is used with dark foreground colors, and the dark primary color with light
+  /// foreground colors.
   final Color backgroundColor;
 
   /// The default text color for text in the circle.
   ///
-  /// Falls back to white if a background color is specified, or the primary
-  /// text theme color otherwise.
+  /// Defaults to the primary text theme color if no [backgroundColor] is
+  /// specified.
+  ///
+  /// If a [backgroundColor] is specified, this falls back to the light primary
```

Comment:
If _no_ [backgroundColor] is specified ... then [foregroundColor] defaults to ...

Reply:
Actually, no, it only defaulted to those if backgroundColor is not null, but I'm rethinking that.  The old code forced black and white as colors, and I don't want to do that, but the way I had it, it would ignore the foreground color you set if you also set the background color, and that's not OK.  I think I fixed it so that if you set both, you get what you set, but if you set one, then we pick a contrasting one for you for the other one.

Reply:
I see, so the foreground color is automatically computed based on the background color. NICE